wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/impl/PersistingProjectRegistry.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.project.server.impl;
+
+import static java.util.Optional.empty;
+import static java.util.stream.Collectors.toSet;
+import static org.eclipse.che.api.fs.server.WsPathUtils.isRoot;
+
+import com.google.common.collect.ImmutableSet;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.eclipse.che.api.core.model.workspace.config.ProjectConfig;
+import org.eclipse.che.api.fs.server.FsManager;
+import org.eclipse.che.api.fs.server.WsPathUtils;
+import org.eclipse.che.api.project.shared.RegisteredProject;
+import org.eclipse.che.api.project.shared.dto.RegisteredProjectDto;
+import org.eclipse.che.dto.server.DtoFactory;
+
```

Comment:
javadoc

Reply:
as I said it is still experimental impl (kinda placeholder) and not bound yet, of course javadoc will be added